website/content/docs/connect/config-entries/service-resolver.mdx

```diff
@@ -1,40 +1,889 @@
+    servicesubset: <subsetName>
+    namespace: <namespaceName>
+    partition: <partitionName>
+    peer: <peerName>
+  failover:                         # requires at least one of the following: service, serviceSubset, namespace, targets, datacenters
+    <localSubsetName>: 
+      targets:
+      - service: <serviceName>
+      - serviceSubset: <serviceSubset>
+      - namespace: <namespaceName>
+      - partition: <partitionName>
+      - datacenter: <datacenterName>
+      - peer: <peerName>
+    `*`: <namespaceName>
+      service: <serviceName>
+      serviceSubset: <serviceSubset>
+      namespace: <namespaceName>
+      datacenters: ["<destinationDatacenter01>", "<destinationDatacenter02>"]
+  loadBalancer:
+    policy: random
+    leastRequestConfig:                # requires policy: leastRequestConfig
+      choiceCount: 2
+    ringHashConfig:                    # requires policy: ringHashConfig
+      minimumRingSize: 1024
+      maximumRingSize: 8192
+    hashPolicies:
+      - field: header                  # cannot specify with SourceIP
+      - fieldValue: <valueToHashOn>    # cannot specify with SourceIP
+      - cookieConfig:
+        session: false
+        ttl: 0s
+        path: <path/to/cookie>
+      - sourceIP: false                # cannot specify with field or fieldValue
+      - terminal: false
+```
+
+</Tab>
+
+</Tabs>
+
+## Specification
+
+This section provides details about the fields you can configure in the configuration entry.
+
+<Tabs>
+
+<Tab heading="HCL" group="hcl">
+
+### `Kind`
+
+Specifies the type of configuration entry to implement. Must be set to `service-resolver`.
+
+#### Values
+
+- Default: None
+- This field is required.
+- Data type: String value that must be set to `service-resolver`.
+
+### `Name`
+
+Specifies a name for the configuration entry. The name is metadata that you can use to reference the configuration entry when performing Consul operations, such as applying a configuration entry to a specific cluster.
+
+#### Values
+
+- Default: None
+- This field is required.
+- Data type: String
+
+### `Namespace` <EnterpriseAlert inline />
+
+Specifies the namespace that the service resolver applies to. Refer to [namespaces](/consul/docs/enterprise/namespaces) for more information.
+
+#### Values
+
+- Default: None
+- Data type: String
+
+### `Partition` <EnterpriseAlert inline />
+
+Specifies the admin partition that the service resolver applies to. Refer to [admin partitions](/consul/docs/enterprise/admin-partitions) for more information.
+
+#### Values
+
+- Default: `default`
+- Data type: String
+
+### `Meta`
+
+Specifies key-value pairs to add to the KV store.
+
+#### Values
+
+- Default: none
+- Data type: Map of one or more key-value pairs
+  - `<KEY>`: String
+  - `<VALUE>`: String or integer
+
+### `ConnectTimeout`
+
+Specifies the timeout duration for establishing new network connections to this service. The duration is measured in seconds.
+
+#### Values
+
+- Default: `0s`
```

Comment:
Interesting question.. I checked out some of our other "0s" default values in the config reference, which led to more questions. In some cases the value of "0s" corresponds to a separate default value (Case 1). In other cases, a value of "0s" disables the feature. 

Case 1- https://developer.hashicorp.com/consul/docs/agent/config/config-files#expiration_leeway
Case 2 - https://developer.hashicorp.com/consul/docs/agent/config/config-files#check_update_interval

Based on this, I think an engineer will need to dive into this function's code in order to get an accurate answer. My guess here is that it's a situation that's more like Case 1 instead of Case 2. My assumption is that 0s means timeout will never occur... but that's just a GUT FEELIN.

Reply:
Found where this lives in the code.. but the comments don't state a "true" default value like the one below it :(
https://github.com/hashicorp/consul/blob/main/agent/structs/config_entry_discoverychain.go#L864-L866